src/components/Sidebar/ActionNode.tsx

```diff
@@ -1,49 +1,212 @@
 import React, { FC, memo, useCallback, ChangeEvent } from 'react';
 import { SidebarButton } from '.';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faPlus } from '@fortawesome/free-solid-svg-icons';
-
+import { faPlus, faCheckCircle } from '@fortawesome/free-solid-svg-icons';
+import { setStateAction } from 'utils/builder';
 import DropDown from 'components/elements/DropDown';
-import { Pathway, State } from 'pathways-model';
+import Input from 'components/elements/Input';
+import { Pathway, GuidanceState, Action } from 'pathways-model';
 import useStyles from './styles';
+import shortid from 'shortid';
 
 const nodeTypeOptions = [
   { label: 'Action', value: 'action' },
   { label: 'Branch', value: 'branch' }
 ];
 
+const actionTypeOptions = [
+  { label: 'Medication Request', value: 'Medication Request' },
+  { label: 'Service Request', value: 'Service Request' },
+  { label: 'Careplan', value: 'Careplan' }
+];
+
+const codeSystemOptions = [
+  { label: 'RXNORM', value: 'http://www.nlm.nih.gov/research/umls/rxnorm' },
+  { label: 'SNOMED', value: 'http://snomed.info/sct' },
+  { label: 'LOINC', value: ' http://loinc.org' }
+];
+
 interface ActionNodeProps {
   pathway: Pathway;
-  currentNode: State;
+  currentNode: GuidanceState;
   changeNodeType: (event: string) => void;
   addNode: (event: string) => void;
+  updatePathway: (pathway: Pathway) => void;
 }
 
-const ActionNode: FC<ActionNodeProps> = ({ pathway, currentNode, changeNodeType, addNode }) => {
+const ActionNode: FC<ActionNodeProps> = ({
+  pathway,
+  currentNode,
+  changeNodeType,
+  addNode,
+  updatePathway
+}) => {
   const styles = useStyles();
   const selectNodeType = useCallback(
     (event: ChangeEvent<{ value: string }>): void => {
       changeNodeType(event?.target.value || '');
     },
     [changeNodeType]
   );
+  const currentNodeKey = currentNode?.key;
+
+  const changeCode = useCallback(
+    (event: ChangeEvent<{ value: string }>): void => {
+      if (!currentNodeKey) return;
+
+      const code = event?.target.value || '';
+      const currentState: GuidanceState = pathway.states[currentNodeKey] as GuidanceState;
+      const action: Action = currentState.action[0];
```

Comment:
We should probably put something in the code style guide to keep everything consistent but you don't need the types here
```suggestion
      const currentState = pathway.states[currentNodeKey] as GuidanceState;
      const action = currentState.action[0];
```